docs/modules/ROOT/pages/reference/extensions/dfdl.adoc

```diff
@@ -0,0 +1,69 @@
+// Do not edit directly!
+// This file was generated by camel-quarkus-maven-plugin:update-extension-doc-page
+[id="extensions-dfdl"]
+= DFDL
+:linkattrs:
+:cq-artifact-id: camel-quarkus-dfdl
+:cq-native-supported: false
+:cq-status: Preview
+:cq-status-deprecation: Preview
+:cq-description: Transforms fixed format data such as EDI message from/to XML using a Data Format Description Language (DFDL).
+:cq-deprecated: false
+:cq-jvm-since: 3.22.0
+:cq-native-since: n/a
+
+ifeval::[{doc-show-badges} == true]
+[.badges]
+[.badge-key]##JVM since##[.badge-supported]##3.22.0## [.badge-key]##Native##[.badge-unsupported]##unsupported##
+endif::[]
+
+Transforms fixed format data such as EDI message from/to XML using a Data Format Description Language (DFDL).
+
+[id="extensions-dfdl-whats-inside"]
+== What's inside
+
+* xref:{cq-camel-components}::dfdl-component.adoc[DFDL component], URI syntax: `dfdl:schemaUri`
+* xref:{cq-camel-components}:dataformats:dfdl-dataformat.adoc[DFDL data format]
+
+Please refer to the above links for usage and configuration details.
+
+[id="extensions-dfdl-maven-coordinates"]
+== Maven coordinates
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-dfdl</artifactId>
+</dependency>
+----
+ifeval::[{doc-show-user-guide-link} == true]
+Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+endif::[]
+
+[id="extensions-dfdl-camel-quarkus-limitations"]
+== Camel Quarkus limitations
+
+While `camel-dfdl` component requires Scala version 2.12, Quarkus BOM enforces Scala version 2.13, which causes `camel-dfdl` to fail. In order to make it work, the application POM needs to override the scala version. You can do that by adding `<dependencyManagement>` in your application POM:
+[source,xml]
```

Comment:
Nor sure this is the usual way to address this. @jamesnetherton any idea ?

Reply:
Unless anyone has any better ideas, I don't see a simple way to resolve this. Maybe only with a custom dfdl BOM that we ask users to import.

Quarkus enforces Scala 2.13 which is not compatible with the Scala version Apache Daffodil was compiled against. We have the same issue with Smooks EDI support.

Reply:
ok, thanks for sharing.